app/Rules/StatusCode404.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Rules;
+
+use App\Services\UrlFetcher;
+use GuzzleHttp\Exception\BadResponseException;
+
+class StatusCode404 extends Rule
+{
+    protected $statusCode;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function check()
+    {
+        $uri = rtrim($this->url, '/').'/page-to-test-404-response';
```

Comment:
I mean, you could test if an URL is passed and if it isn't, check for the ramdom one.